browser/ephemeral_storage/ephemeral_storage_tab_helper.cc

```diff
@@ -124,10 +130,31 @@ void EphemeralStorageTabHelper::CreateEphemeralStorageAreasForDomainAndURL(
                 kSessionStorageSuffix))
           : base::nullopt);
 
+  if (base::FeatureList::IsEnabled(
+          net::features::kBraveEphemeralStorageKeepAlive)) {
+    // keep the ephemeral storage alive for some time to handle redirects
+    // including meta refresh or other page driven "redirects" that end up back
+    // at the original origin
+    base::ThreadTaskRunnerHandle::Get()->PostDelayedTask(
+        FROM_HERE,
+        base::BindOnce([](scoped_refptr<content::TLDEphemeralLifetime>
+                              tld_ephemeral_lifetime) {},
+                       tld_ephemeral_lifetime_),
+        g_storage_keep_alive_for_testing.is_min()
+            ? kStorageKeepAliveDelay
+            : g_storage_keep_alive_for_testing);
+  }
+
   tld_ephemeral_lifetime_ = content::TLDEphemeralLifetime::GetOrCreate(
```

Comment:
so each time an instance will be the same, we still want to do ```PostDelayedTask```?
I think it's better to do ```TLDEphemeralLifetime::GetOrCreate```, check if we got a different instance and then run a timer with an old instance if we need to.

Reply:
That gets very complicated if you have multiple tabs open and different types of redirects (like meta refresh). This is much simpler and more reliable because the the ephemeral storage is only destroyed when the scoped refptr count gets to zero so we don't have to worry about debouncing across requests and tabs

Reply:
also this only gets called when the storage changes https://github.com/brave/brave-core/pull/8351/files#diff-c50fbda1dae121ba41d87ae616ec4ea89acb79ed5465918fc9ddaa3301e0b6f2R82 but the effect would be the same either way. We want to keep a reference for 30 seconds longer than it would normally live based on the refcount

Reply:
my proposal is just about this:
```
auto new_tld_ephemeral_lifetime = content::TLDEphemeralLifetime::GetOrCreate(...);
if (new_tld_ephemeral_lifetime != tld_ephemeral_lifetime_) {
  PostTask(...);
  tld_ephemeral_lifetime_ = new_tld_ephemeral_lifetime;
}
```
but now I see that we don't recreate the instance if a domain hasn't changed, so the current implementation is fine.